be/src/exprs/agg/retention.h

```diff
@@ -0,0 +1,129 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <cstring>
+#include <limits>
+#include <type_traits>
+
+#include "column/array_column.h"
+#include "column/binary_column.h"
+#include "column/datum.h"
+#include "column/fixed_length_column.h"
+#include "column/type_traits.h"
+#include "column/vectorized_fwd.h"
+#include "exprs/agg/aggregate.h"
+#include "gen_cpp/Data_types.h"
+#include "glog/logging.h"
+#include "gutil/casts.h"
+#include "runtime/mem_pool.h"
+#include "thrift/protocol/TJSONProtocol.h"
+#include "udf/udf_internal.h"
+#include "util/phmap/phmap_dump.h"
+#include "util/slice.h"
+
+namespace starrocks::vectorized {
+struct RetentionState {
+    void merge_array_element(const DatumArray& datum_array) {
+        auto size = datum_array.size();
+        if (!is_initial) {
+            is_initial = true;
+            boolean_vector.resize(size);
+        }
+
+        // All array conditions' size is equal.
+        DCHECK_EQ(boolean_vector.size(), datum_array.size());
```

Comment:
I'm not sure where to ensure that all the input arrays are the same size and length

Reply:
> I'm not sure where to ensure that all the input arrays are the same size and length

To the definition of retention, the size is just count of conditions

Reply:
It looks like a crash will occur if the input array does not match

Reply:
> It looks like a crash will occur if the input array does not match

Actually It must match. This check is unneeded, I will remove it.

Reply:
t looks like we can't stop a query like this
```
select retention(array_col) from data;
```

array_col data was:
```
[true,true,true]
[true,true]
```

which may cause a crash